plugins/outputs/bigquery/bigquery.go

```diff
@@ -0,0 +1,215 @@
+package bigquery
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"reflect"
+
+	"cloud.google.com/go/bigquery"
+	"github.com/influxdata/telegraf"
+	"golang.org/x/oauth2/google"
+	"google.golang.org/api/option"
+)
+
+const (
+	testingHostEnv     = "BIGQUERY_TESTING_HOST"
+	defaultOffSetKey   = "offset-key.json"
+	timeStampFieldName = "timestamp"
+)
+
+type BigQuery struct {
+	CredentialsFile string `toml:"credentials_file"`
+	Project         string `toml:"project"`
+	Dataset         string `toml:"dataset"`
+
+	client *bigquery.Client
+}
+
+var sampleConfig = `	
+  ## Credentials File
+  credentials_file = "/path/to/service/account/key.json"
+  ## GCP Project
+  project = "my-gcp-project"
+
+  ## The namespace for the metric descriptor
+  dataset = "telegraf"
+`
+
+func (b *BigQuery) Connect() error {
+	if b.Project == "" {
+		return fmt.Errorf("Project is a required field for BigQuery output")
+	}
+
+	if b.Dataset == "" {
+		return fmt.Errorf("Dataset is a required field for BigQuery output")
+	}
+
+	b.setUpClient()
+
+	return nil
+}
+
+func (b *BigQuery) setUpClient() error {
+	if endpoint, present := os.LookupEnv(testingHostEnv); present {
+		return b.setUpTestClient(endpoint)
+	}
+
+	return b.setUpDefaultClient()
+}
+
+func (b *BigQuery) setUpTestClient(endpoint string) error {
+	noAuth := option.WithoutAuthentication()
+	endpoints := option.WithEndpoint("http://" + endpoint)
+
+	ctx := context.Background()
+
+	if c, err := bigquery.NewClient(ctx, b.Project, noAuth, endpoints); err != nil {
+		return err
+	} else {
+		b.client = c
+		return nil
+	}
+}
+
+func (b *BigQuery) setUpDefaultClient() error {
+	var credentialsOption option.ClientOption
+
+	ctx := context.Background()
+
+	if b.CredentialsFile != "" {
+		credentialsOption = option.WithCredentialsFile(b.CredentialsFile)
+	} else {
+		creds, err := google.FindDefaultCredentials(ctx)
+		if err != nil {
+			return fmt.Errorf(
+				"unable to find GCP Application Default Credentials: %v."+
+					"Either set ADC or provide CredentialsFile config", err)
+		}
+		credentialsOption = option.WithCredentials(creds)
+	}
+
+	client, err := bigquery.NewClient(ctx, b.Project, credentialsOption)
+	b.client = client
+	return err
+}
+
+// Write the metrics to Google Cloud BigQuery.
+func (b *BigQuery) Write(metrics []telegraf.Metric) error {
+	groupedMetrics := b.groupByMetricName(metrics)
+
+	for k, v := range groupedMetrics {
+		if err := b.insertToTable(k, v); err != nil {
+			return err
```

Comment:
This means you drop all remaining metrics. Skipping over this erroneous one but process the others might be better
```suggestion
			b.Log.Errorf("inserting metric %q failed: %v", <metric id/name>, err)
			continue
```